### ISSUE TITLE
Request attribute name for the session instance

### ISSUE THREAD

**Issue by pine3ree** [zendframework/zend-expressive-session#31]
Wouldn' it be better to set (namespaces omitted) `SessionMiddleware::ATTR_SESSION` to `SessionInterface::class`? This would prevent accidental attribute overwrites and it's also more meaninful about the purpose, liks it's done for RouteResult. I believe that is a generally good convention to use interface class names to identify objects stored in request attributes.
kind regards

**Comment by geerteltink** [comment]
> I believe that is a generally good convention to use interface class names to identify objects stored in session attributes.

I guess you mean ... stored in **request** attributes.
And yes I agree.

**Comment by weierophinney** [comment]
The reason we didn't go this route is due to prior art.

A number of other session libraries already exist, and most have standardized on the attribute "session" — which means other middleware and handlers that interact with sessions are looking for that particular attribute name as well. To make it possible to switch implementations without needing to fix every case where those attributes are being accessed, we chose to do likewise.

Our compromise was to define this in a class constant, as doing so allows users to use autocompletion, and users will also get errors if they misspell the constant name (which is not true if you misspell a generic string).

I definitely generally agree on using class or interface names for these purposes; this, however, was a place where we had to make a different choice.
